src/store/selectors/exchange.ts

```diff
@@ -384,4 +398,160 @@ export const getHaltedState = createSelector(state, allState => {
   return allState.halted
 })
 
+export interface UserVaults extends VaultEntry {
+  borrowed: string
+  collateral: string
+  deposited: number
+  depositedSign: string
+  cRatio: string
+  minCRatio: number
+  currentDebt: number
+  currentDebtSign: string
+  maxBorrow: string
+  interestRate: string
+  liquidationPrice: string
+}
+
+export const getUserVaults = createSelector(
+  vaults,
+  userVaults,
+  synthetics,
+  assetPrice,
+  (allVaults, allUserVaults, allSynthetics, allAssetPrice) => {
+    const vaultData: UserVaults[] = []
+    Object.values(allUserVaults).forEach(userVault => {
+      const currentVault = allVaults[userVault.vault.toString()]
+      if (
+        typeof allAssetPrice[currentVault.collateral.toString()] !== 'undefined' &&
+        typeof allAssetPrice[currentVault.synthetic.toString()] !== 'undefined'
+      ) {
+        const cRatioCalculated = calculateCRatio(
+          allAssetPrice[currentVault.synthetic.toString()].val,
+          allSynthetics[currentVault.synthetic.toString()].supply.scale,
+          allAssetPrice[currentVault.collateral.toString()].val,
+          currentVault.collateralAmount.scale,
+          userVault.syntheticAmount.val,
+          userVault.collateralAmount.val
+        )
+        const interestRate =
+          Number(printBN(currentVault.debtInterestRate.val, currentVault.debtInterestRate.scale)) *
+          100
+
+        const openFeeBN = stringToMinDecimalBN(
+          (Number(printBN(currentVault.openFee.val, currentVault.openFee.scale)) * 100).toString()
+        )
+
+        const maxBorrow = calculateAmountBorrow(
+          allAssetPrice[currentVault.synthetic.toString()].val,
+          allSynthetics[currentVault.synthetic.toString()].supply.scale,
+          allAssetPrice[currentVault.collateral.toString()].val,
+          currentVault.collateralAmount.scale,
+          userVault.collateralAmount.val,
+          Number(
+            Math.pow(
+              Number(
+                printBN(currentVault.collateralRatio.val, currentVault.collateralRatio.scale)
+              ) / 100,
+              -1
+            )
+          ).toString()
+        )
+          .sub(userVault.syntheticAmount.val)
+          .mul(new BN(10).pow(new BN(openFeeBN.decimal + 2)))
+          .div(new BN(10).pow(new BN(openFeeBN.decimal + 2)).add(openFeeBN.BN))
+        vaultData.push({
+          ...userVault,
+          borrowed: addressToAssetSymbol[currentVault.synthetic.toString()] || 'XYZ',
+          collateral: addressToAssetSymbol[currentVault.collateral.toString()] || 'XYZ',
+          deposited: Number(
+            printBN(userVault.collateralAmount.val, userVault.collateralAmount.scale)
+          ),
+          depositedSign: addressToAssetSymbol[currentVault.collateral.toString()] || 'XYZ',
+          cRatio: cRatioCalculated !== 'NaN' ? transformBN(cRatioCalculated) : 'NaN',
```

Comment:
this stuff should be done in component not selector

Reply:
ok but transformBN ('NaN') is not possible